src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,221 @@
+    * bounding box.
+    *
+    * @param globalOrigin Coordinates of the global origin
+    * @param lowerLeftCorner Lower left corner of the bounding box
+    * @param joints Array containing the input data representing joint sets
+    * @return Seq of "master" joints that represent input joint sets. These will be used as a starting point
+    *         to generate full joint set.
+    */
+  private def findMasterJoints(globalOrigin: Array[Double], lowerLeftCorner: Array[Double],
+                               joints: Seq[Array[Double]]): Option[Seq[Joint]] = {
+    val masterJoints = joints flatMap { parameters =>
+      if (parameters.length == 6) {
+        // TODO: Implement non-persistent joint generator
+        val normal = findJointNormal(parameters(0), parameters(1))
+        Some(Joint(normal, globalOrigin, center=Array(lowerLeftCorner(0), lowerLeftCorner(1), lowerLeftCorner(2)),
+          phi=parameters(4), cohesion=parameters(5), shape=Vector.empty))
+      } else {
+        // TODO: Implement stochastic generator
+        println("Error, Joint Input Data: Stochastic joint generation not implemented yet")
+        None
+      }
+    }
+    if (masterJoints.length != joints.length) {
+      None
+    } else {
+      Some(masterJoints)
+    }
+  }
+}
+
+/**
+  * Provides functionality to generate joint sets for input joint parameters and faces
+  * that represent faces of input rock volume.
+  *
+  * @param globalOrigin Coordinates of the global origin
+  * @param boundingBox Coordinates specifying the lower left and upper right corners of the box that bounds
+  *                    the domain of interest.
+  * @param rockVolumeData Array of arrays describing each of the faces that define the boundaries of
+  *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+  *                       y and z coordinates of a point that lies within the plane containing the face.
+  *                       Strike and dip should be specified in degrees.
+  * @param jointSetData Array of arrays containing the input data representing joint sets. The inputs in each array
+  *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+  */
+case class JointGenerator(globalOrigin: Array[Double], boundingBox: Array[Double], rockVolumeData: Seq[Array[Double]],
+                     jointSetData: Seq[Array[Double]]) {
+  val origin = globalOrigin
+  val lowerLeftCorner = Array(boundingBox(0), boundingBox(1), boundingBox(2))
+  val upperRightCorner = Array(boundingBox(3), boundingBox(4), boundingBox(5))
+  val rockVolume = JointGenerator.findBoundingFaces(globalOrigin, rockVolumeData)
+  val masterJoints = JointGenerator.findMasterJoints(globalOrigin, lowerLeftCorner, jointSetData).get
+
+  if (masterJoints.isEmpty) {
+    throw new IllegalArgumentException("ERROR: Stochastic joint generation not yet implemented")
```

Comment:
This is exactly like what I had in mind above. `IllegalArgumentException` works but `UnsupportedOperationException` might be a bit more standard.